### DISCUSSION TITLE
How to Access Attachments and Their Content in Budibase Screens?

### DISCUSSION THREAD

**Discussion by Abd-Atif** [Budibase/budibase#12394] [Help] [answered]
Hi Budibase Community,

I'm currently working on a project using Budibase and have encountered a challenge. I need some guidance on how to access attachments and view their content within a screen.

Could anyone provide insights or steps on how to achieve this? I am particularly interested in understanding:

**How to get content of .csv files stored in budibase db and how to process that (to convert in base64) ?
How to display attachments (like images or documents) in a Budibase screen.
Methods to interact with these attachments, such as viewing or downloading them.**

Any relevant documentation or examples that might assist in this process.

Any help or pointers would be greatly appreciated. Thank you in advance for your support!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Abd-Atif,

The image component is what would be used to display images within the design view within Budibase. Related documentation [here](https://docs.budibase.com/docs/image).

- You can import your CSV directly into a table using the import button on which ever selected data source you have chosen to use. The [data provider](https://docs.budibase.com/docs/data-provider) component within the design area also has the option to import a CSV into it for quick and easy access to your data.
- This might help for downloading your attachments. https://github.com/Budibase/budibase/discussions/11350
- This might help for viewing your attachments. This will only work with images as thats what the image component is equipped to handle. https://discord.com/channels/733030666647765003/1153069081029787759/1153069081029787759
- As for your base 64 question, you'd likely have to do something with [automations](https://docs.budibase.com/docs/automation-steps)/[custom automation](https://docs.budibase.com/docs/custom-automations) steps in order to output the base 64 of the csv and store this within a table.

I hope this points you in the right direction.

**Reply by ecmonsen** [reply]
Hey this is a great summary! For your last point about automations - what's the best way to pass the attachment's data to a REST API (or to have the external API service download the attachment)?
